x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -177,10 +177,22 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 			continue
 		}
 
-		// retrieve all the details for all the metrics available in the current region
-		listMetricsOutput, err := aws.GetListMetricsOutput("*", regionName, m.Period, m.IncludeLinkedAccounts, m.OwningAccount, m.MonitoringAccountID, svcCloudwatch)
-		if err != nil {
-			m.Logger().Errorf("Error while retrieving the list of metrics for region %s: %w", regionName, err)
+		// retrieve all the details for all the metrics available in the current region when no namespace is specified
+		// otherwise only retrieve metrics from the specific namespaces from the config
+		var listMetricsOutput []aws.MetricWithID
+		if len(namespaceDetailTotal) == 0 {
+			listMetricsOutput, err = aws.GetListMetricsOutput("*", regionName, m.Period, m.IncludeLinkedAccounts, m.OwningAccount, m.MonitoringAccountID, svcCloudwatch)
+			if err != nil {
+				m.Logger().Errorf("Error while retrieving the list of metrics for region %s and namespace %s: %w", regionName, "*", err)
```

Comment:
Do we need to include the namespace in this error message?

Reply:
Also I think that this is can be a good integration test to include it. WDYT?

Reply:
The error message here shows namespace is `*` because in this if statement we are calling `ListMetrics` with `*` all namespaces. The else statement part is the one with a specific namespace.

Reply:
@gizas, most integration tests start the metricset/input and check that they produce a non-zero number of events.

What kind of check do you have in mind?

 What do you think about creating a dedicated issue to enhance the integration tests for the cloudwatch metricset?

Reply:
@kaiyan-sheng, this change may increase the number of API calls because we're making one API call for each namespace in the config. 

However, it shouldn't make a difference since I don't expect configurations to have more than 1-3 namespaces, right?

Reply:
@zmoog https://github.com/elastic/beats/issues/41146 FYI

Reply:
@zmoog Yeah that's the original thought when we decide to use "*" instead of individual namespace in this API call. But I think at this point, the benefit of using specific namespaces with a higher number of API calls wins. 
I will update the document here: https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-aws.html#aws-api-requests

Reply:
We actually already have tests around this but since we are mocking the ListMetrics API and what's returning from the API, the test is no use for distinguish namespaces. As we discussed earlier, I will merge this PR to get the fix in but look into the test later. Thanks!

Reply:
@zmoog I tested the pr again with the config below to double check how many times this ListMetrics API will be called. Turned out we are calling it once because namespaces are the same in both parts under metrics:
```
- module: aws
  period: 60s
  metricsets:
    - cloudwatch
  data_granularity: 60s
  include_linked_accounts: true
  regions:
    - ap-southeast-1
  metrics:
    - dimensions:
        - name: CanaryName
          value: canary1
      namespace: CloudWatchSynthetics
    - dimensions:
        - name: CanaryName
          value: canary2
      namespace: CloudWatchSynthetics
```